api/src/diag/types.ts

```diff
@@ -96,3 +96,37 @@ export enum DiagLogLevel {
 export interface ComponentLoggerOptions {
   namespace: string;
 }
+
+export interface LoggerOptions {
+  /**
+   * The {@link DiagLogLevel} used to filter logs sent to the logger.
+   *
+   * @defaultValue DiagLogLevel.INFO
+   */
+  logLevel?: DiagLogLevel;
+
+  /**
+   * Setting this value to `true` will suppress the warning message normally emitted when registering a logger when another logger is already registered.
+   */
+  suppressOverrideMessage?: boolean;
+}
+
+export interface DiagLoggerApi {
+  /**
+  * Set the global DiagLogger and DiagLogLevel.
+  * If a global diag logger is already set, this will override it.
+  *
+  * @param logger - [Optional] The {@link DiagLogger} instance to set as the default logger.
```

Comment:
```suggestion
  * @param logger - The {@link DiagLogger} instance to set as the default logger.
```

Reply:
Typedoc actually understands the optionality so yes the text is redundant